### PR TITLE
[SPARK-44595][CONNECT] Make the user session cache number and cache time be configurable in spark connect service

### PR DESCRIPTION
**Pull request by zhaomin1423**
### What changes were proposed in this pull request?

Added configs for user session cache size and cache timeout in connect service.

### Why are the changes needed?

Now, the cache size of user session is 100, the cache timeout is 3600. Make them modifiable to meet diverse scenarios.

### Does this PR introduce _any_ user-facing change?

no
### How was this patch tested?

Existing can test it.

### CODE REVIEW

**Review by Hisoka-X — COMMENTED** [review]
We can avoid unnecessary multiline strings.

**Comment by zhaomin1423** [comment]
> We can avoid unnecessary multiline strings.

Thanks, updated.

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala [hunk 1]
```diff
@@ -121,4 +121,18 @@ object Connect {
       .version("3.5.0")
       .booleanConf
       .createWithDefault(false)
+
+  val CONNECT_USER_SESSION_CACHE_SIZE =
+    ConfigBuilder("spark.connect.userSession.cacheSize")
```
Comment (beliefer) [inline connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala:126]: The Spark Sessions are shared common or for user?

Comment (zhaomin1423) [reply connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala:126]: It is common

**Comment by zhaomin1423** [comment]
Could you help me review it? @LuciferYang

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala (hunk 1 above)
Comment (beliefer) [reply connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala:126]: So, how about `spark.connect.session.cacheSize` ?

Comment (zhaomin1423) [reply connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala:126]: It is ok, thanks

**Review by juliuszsompolski — CHANGES_REQUESTED** [review]

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala [hunk 2]
```diff
@@ -121,4 +121,18 @@ object Connect {
...
+      .version("4.0.0")
+      .intConf
+      .createWithDefault(100)
+
+  val CONNECT_USER_SESSION_CACHE_TIMEOUT =
+    ConfigBuilder("spark.connect.userSession.cacheTimeout")
+      .doc("Sets the cache timeout of user session.")
+      .version("4.0.0")
+      .intConf
+      .createWithDefault(3600)
```
[the remark this answers is not in the document]
Comment (zhaomin1423) [reply connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala:137]: Thanks for your input, look forward to your work.

**Pull request closed without merging**